test/androidTest/java/org/thoughtcrime/securesms/util/ConversationActions.java

```diff
@@ -69,141 +53,158 @@ public void conversationListActivityTest() {
                         childAtPosition(
                                 childAtPosition(
                                         withClassName(is("android.widget.ScrollView")),
                                         0),
                                 1)));
-        appCompatEditText4.perform(scrollTo(), replaceText("test"), closeSoftKeyboard());
+        appCompatEditText.perform(scrollTo(), replaceText(name), closeSoftKeyboard());
 
-        ViewInteraction appCompatButton2 = onView(
+        ViewInteraction appCompatButton = onView(
                 allOf(withId(android.R.id.button1), withText("OK"),
                         childAtPosition(
                                 childAtPosition(
                                         withId(R.id.buttonPanel),
                                         0),
                                 3)));
-        appCompatButton2.perform(scrollTo(), click());
+        appCompatButton.perform(scrollTo(), click());
+    }
 
-        ViewInteraction appCompatImageButton = onView(
-                allOf(withContentDescription("Navigate up"),
-                        childAtPosition(
-                                allOf(withId(R.id.toolbar),
-                                        childAtPosition(
-                                                withId(R.id.collapsing_toolbar),
-                                                3)),
-                                1),
+    public static void pinLastMessage() {
+        longPressMessageAt(0);
+
+        ViewInteraction actionMenuItemView = onView(
+                allOf(withId(R.id.menu_context_pin_message), withContentDescription("Pin message"),
                         isDisplayed()));
-        appCompatImageButton.perform(click());
+        actionMenuItemView.perform(click());
+    }
 
-        ViewInteraction textView2 = onView(
-                allOf(withId(R.id.title), withText("test"),
-                        childAtPosition(
-                                allOf(withId(R.id.content),
-                                        childAtPosition(
-                                                IsInstanceOf.<View>instanceOf(android.widget.RelativeLayout.class),
-                                                2)),
-                                0),
+    public static void unpinLastMessage() {
+        longPressMessageAt(0);
+
+        ViewInteraction actionMenuItemView = onView(
+                allOf(withId(R.id.menu_context_unpin_message), withContentDescription("Unpin message"),
+                        isDisplayed()));
+        actionMenuItemView.perform(click());
+    }
+
+    public static void markAsUnread() {
+        ViewInteraction actionMenuItemView = onView(
+                allOf(withId(R.id.menu_context_mark_as_unread),
                         isDisplayed()));
-        textView2.check(matches(withText("test")));
+        actionMenuItemView.perform(click());
+    }
+
+    public static void longPressMessageAt(int i) {
```

Comment:
could i be more descriptive.

Reply:
How so? It's long pressing a specific message based off of the index, I can change the i variable but I don't see how the function name can be more specific since we already know it's acting on messages in the conversation view.

Reply:
I was thinking the `int i` could be more descriptive as in position or index but I guess its fine. Just thinking if I see `i` in intellisense as a variable I need to give. I'll be confused.

Reply:
I read your message as "could it be more descriptive", my bad. Yeah I'll change i to something else.